File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int pro = 1;
        int zero = 0;
        vector<int> prod;

        for (int i : nums) {
            if (i != 0)
                pro *= i;
            else
                zero++;
        }

        if(zero>1)
        return vector<int>(nums.size(),0) ;
        for(int i : nums)
        {
            if(zero)
            {
                if(i==0)
                prod.push_back(pro) ;
                else
                prod.push_back(0) ;
            }
            else 
            prod.push_back(pro/i) ;
        }

        return prod ;
       
    }
};

File: Data Structures & Algorithms/sort-an-array/submission-2.py
class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:
        for i in range(len(nums)):
            swaped =False
            for  j in range(len(nums)-i-1):
                if(nums[j]>nums[j+1]):
                    nums[j] , nums[j+1] = nums[j+1], nums[j]
                    swaped  =  True 
            if swaped==False:
                break 
        return nums 
